### PR TITLE
Dense one-to-one Hough transform and Point tracking in the Hough space

### PR DESCRIPTION
**Pull request by zyannick**
I proposed a fast computation of dense one to one Hough transform and a method to track lines in the Hough space

### CODE REVIEW

**Comment by zyannick** [comment]
IT seems there is an error somewhere!
I have some trouble to commit in the office, I'm commit another version at this evening

File: vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/fast_hough.hh [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+
+using namespace vpp;
+using namespace std;
+using namespace Eigen;
+using namespace iod;
+using namespace cv;
+
+namespace vpp{
+
```
Comment (matt-42) [inline vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/fast_hough.hh]: It's not clear which function are part of the API you're providing and which one are only internal.

File: vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/fast_hough.hpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+    return T;
+}
+
+cv::Mat accumulatorToFrame(std::list<vint2> interestedPoints, int rhomax, int T_theta)
+{
+    Mat T = Mat(int(rhomax),int(T_theta),CV_8UC1,cvScalar(0));
+    int r = 0;
+    for(auto& ip : interestedPoints)
```
Comment (matt-42) [inline vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/fast_hough.hpp]: Use const auto& ip or auto ip if you do not need write access to this variable.

File: examples/hough_extruder.cc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+#include <chrono>
+//1
+
+using namespace  std;
+using namespace Eigen;
+using namespace cv;
+using namespace vpp;
+using namespace std::chrono;
+
+int main(int argc, char *argv[])
```
[the remark this answers is not in the document]
Comment (zyannick) [reply examples/hough_extruder.cc]: Okay I will work on comments this week end

File: vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/dense_one_to_one_hough.hh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+using namespace Eigen;
+using namespace iod;
+using namespace cv;
+
+namespace vpp{
+
+#define Two_PI 2*M_PI
+
+
+std::list<vint2> Hough_Lines_Parallel(image2d<uchar> img,
```
[the remark this answers is not in the document]
Comment (zyannick) [reply vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/dense_one_to_one_hough.hh]: Okay. Globally I need to explain more what do the functions. 
Thank you very much.
I`ll use my week end on this way

[the remark this answers is not in the document]
Comment (matt-42) [reply vpp/algorithms/Line_tracker_4_sfm/Fast_DHT/dense_one_to_one_hough.hh]: Yes, and put together the functions that you want people to use: 
Check video extruder for example:
https://github.com/matt-42/vpp/blob/master/vpp/algorithms/video_extruder.hh
There is only 2 functions in the main header and the internal machinary is hidden inside the video_extruder directory.
You could add a header algorithms/line_tracker_for_sfm.hh with the signature of the functions people need to know about. Nothing more.

File: examples/hough_extruder.cc [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+{
+    high_resolution_clock::time_point t1 = high_resolution_clock::now();
+    Fast_DHT_Matching(feature_matching,Theta_max::SMALL, Sclare_rho::SAME,
```
[the remark this answers is not in the document]
Comment (zyannick) [reply examples/hough_extruder.cc]: Hi , 
The features I am tracking are in the file feature_matching_hough.hh. I've slightly modified the api

**Pull request merged**